nemo_curator/modules/semantic_dedup/clusteringmodel.py

```diff
@@ -122,36 +122,39 @@ def __call__(self, embeddings_dataset: DocumentDataset) -> DocumentDataset:  # n
             if self.clustering_input_partition_size is not None:
                 embeddings_df = embeddings_df.repartition(partition_size=self.clustering_input_partition_size)
 
-            try:
-                embeddings_df = embeddings_df.to_backend("pandas").persist()
-                embeddings_length = embeddings_df.shape[0].compute()
-
-                if embeddings_length < self.n_clusters:
-                    msg = (
-                        "Number of clusters is greater than the number of documents in your dataset: "
-                        f"dataset length is {embeddings_length} while n_clusters is set to {self.n_clusters}. "
-                        f"Please reduce n_clusters to be less than or equal to {embeddings_length}."
-                    )
-                    raise ValueError(msg)
-            except IndexError as e:
-                msg = (
-                    f'Original error message: "{e}". '
-                    "This could be due to empty partitions in your DocumentDataset. "
-                    "Please check your dataset for empty partitions and remove them if necessary."
-                )
-                raise IndexError(msg) from e
-
-            embeddings_df = embeddings_df.to_backend("cudf")
+            # We need this optimize call since the map_partitions(get_array_from_df) seems to be returning different
+            # number of partitions
+            embeddings_df = embeddings_df.optimize()
```

Comment:
I was wondering if you could explain more why this is needed?

Reply:
Added a comment re the same, if its not clear lmk